test/metabase/api/card_test.clj

```diff
@@ -2450,3 +2450,21 @@
           (is (not (contains? (task.persist-refresh/job-info-for-individual-refresh)
                               (u/the-id parchived)))
               "Scheduled refresh of archived model"))))))
+
+(deftest template-tag-referenced-card-ids-test
+  (testing "template-tag-referenced-card-ids returns the correct card ids from the query"
+    (mt/with-temp Card [card {:dataset_query {:database (mt/id)
+                                              :type     :native
+                                              :native   {:query         "select * from {{#14}} as a join {{#15}} as b on a.id = b.id"
+                                                         :template-tags {"#14" {:id           "some id"
```

Comment:
I'd try to add another template tag that is not a card for this test to make sure we're not accidentally returning "everything in the template tags"